qt/widgets/common/inc/MantidQtWidgets/Common/FitScriptGeneratorDataTable.h

```diff
@@ -0,0 +1,122 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "DllOption.h"
+#include "MantidQtWidgets/Common/IndexTypes.h"
+
+#include <string>
+#include <vector>
+
+#include <QEvent>
+#include <QModelIndex>
+#include <QObject>
+#include <QPainter>
+#include <QPersistentModelIndex>
+#include <QString>
+#include <QStyleOptionViewItem>
+#include <QStyledItemDelegate>
+#include <QTableWidget>
+#include <QTableWidgetItem>
+
+namespace MantidQt {
+namespace MantidWidgets {
+
+/**
+ * This class represents the table widget which holds domain data for the
```

Comment:
is this similar to the one in indirect?

Reply:
Somewhat similar, however it does not have a `Mask X Range` column. It also does a few extra things like allows multiple selection and formats the data inside a bit better.

Reply:
would it be worth using this in the indirect GUI?